test/utils/key-camelizer.test.js

```diff
@@ -0,0 +1,165 @@
+'use strict'
+
+const expect = require('chai').expect
+const { camelize, decamelize } = require('../../app/utils/key-camelizer')
+
+const simpleSnakeObj = {
+  attr_one: 'foo',
+  attr_two: 'bar'
+}
+
+const simpleCamelObj = {
+  attrOne: 'foo',
+  attrTwo: 'bar'
+}
+
+const simplePascalObj = {
+  AttrOne: 'foo',
+  AttrTwo: 'bar'
+}
+
+const complexSnakeObj = {
+  attr_one: 'foo',
+  attr_two: {
+    nested_attr1: 'bar'
+  },
+  attr_three: {
+    nested_attr2: {
+      nested_attr3: [{
+        nested_in_array1: 'baz'
+      }, {
+        nested_in_array2: 'hello'
+      }, {
+        nested_in_array3: ['world', 'boo']
+      }]
+    }
+  }
+}
+
+const complexCamelObj = {
+  attrOne: 'foo',
+  attrTwo: {
+    nestedAttr1: 'bar'
+  },
+  attrThree: {
+    nestedAttr2: {
+      nestedAttr3: [{
+        nestedInArray1: 'baz'
+      }, {
+        nestedInArray2: 'hello'
+      }, {
+        nestedInArray3: ['world', 'boo']
+      }]
+    }
+  }
+}
+
+const complexPascalObj = {
+  AttrOne: 'foo',
+  AttrTwo: {
+    NestedAttr1: 'bar'
+  },
+  AttrThree: {
+    NestedAttr2: {
+      NestedAttr3: [{
+        NestedInArray1: 'baz'
+      }, {
+        NestedInArray2: 'hello'
+      }, {
+        NestedInArray3: ['world', 'boo']
+      }]
+    }
+  }
+}
+
+const complexCustomObj = {
+  'attr-one': 'foo',
+  'attr-two': {
+    'nested-attr1': 'bar'
+  },
+  'attr-three': {
+    'nested-attr2': {
+      'nested-attr3': [{
+        'nested-in-array1': 'baz'
+      }, {
+        'nested-in-array2': 'hello'
+      }, {
+        'nested-in-array3': ['world', 'boo']
+      }]
+    }
+  }
+}
+
+const nestedNullObj = {
+  attr_one: 'foo',
+  attr_two: 'bar',
+  attr_three: {
+    nested_obj: null
+  }
+}
+
+const camelizedNestedNullObj = {
+  attrOne: 'foo',
+  attrTwo: 'bar',
+  attrThree: {
+    nestedObj: null
+  }
+}
+
+describe('camelize', () => {
```

Comment:
there's a lot of repetition here, these tests could be [parameterised](https://mochajs.org/#dynamically-generating-tests) to DRY things up